### PR TITLE
chore(abci): set NextBlockDelay to 1s (default)

### PR DESCRIPTION
**Pull request by melekes**
follow-up to https://github.com/cometbft/cometbft/pull/4339

otherwise, CometBFT spits out blocks like crazy 😸

### CODE REVIEW

**Comment by cason** [comment]
`test/app/test.sh: line 15: abci-cli: command not found`

**Comment by cason** [comment]
Can you also change that on the `e2e` application? Or is it already fixed?

**Comment by melekes** [comment]
> Can you also change that on the `e2e` application? Or is it already fixed?

already fixed

**Review by andynog — APPROVED** [review]
lgtm, e2e would also be good

By the way, could not find any information about it on https://docs.cometbft.com/v1.0/spec/abci/abci++_app_requirements

**Comment by andynog** [comment]
> > Can you also change that on the `e2e` application? Or is it already fixed?
> 
> already fixed

Yes here https://github.com/cometbft/cometbft/blob/732e2dfc23e30b6275c3f97bd7735751c6806eb9/test/e2e/app/app.go#L480

**Comment by andynog** [comment]
@melekes for some reason the there are a couple of tests that are failing. I've re-run then a few times and they continue to fail so it's worth investigate.

**Review by wesl-ee — APPROVED** [review]
makes sense. in the absence of a defined NextBlockDelay the fallback is what's defined in the config, which defaults to 0 even when unset as this field is deprecated

**Comment by melekes** [comment]
> @melekes for some reason the there are a couple of tests that are failing. I've re-run then a few times and they continue to fail so it's worth investigate.

Maybe we're using `kvstore` app in those, and `NextBlockDelay` expected to be `0`.

File: abci/example/kvstore/kvstore.go [hunk 1]
```diff
@@ -115,6 +119,13 @@ func (app *Application) SetGenBlockEvents() {
 	app.genBlockEvents = true
 }
 
+// SetNextBlockDelay sets the delay for the next finalized block. Default is 0
+// here because kvstore is mostly used for testing. In production, the default
+// is 1s.
```
Comment (cason) [inline abci/example/kvstore/kvstore.go:124]:
```suggestion
// is 1s, mimicing the default for the deprecated `timeout_commit` parameter.
```

File: proxy/client.go [hunk 2]
```diff
@@ -263,20 +265,36 @@ func (r *remoteClientCreator) newABCIClient() (abcicli.Client, error) {
 // "_connsync" variant (i.e. "kvstore_connsync", etc.), which attempts to
 // replicate the same concurrency model as the remote client.
 func DefaultClientCreator(addr, transport, dbDir string) ClientCreator {
+	// Default is zero for kvstore and persistent_kvstore.
+	// Set to 1s to mimic the real world app.
```
Comment (cason) [inline proxy/client.go:269]:
```suggestion
	// Replaces deprecated `timeout_commit` parameter.
	// Set to 1s to mimic the real world app.
```

**Review by cason — APPROVED** [review]
+1

**Pull request merged**